src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java

```diff
@@ -150,6 +165,77 @@ private void testCreateModuleWithNonExistName(String name) {
+    }
+
+    @Test
+    public void testCreateObjectFromFullModuleNames() throws Exception {
+        final Method method = factory.getClass().getDeclaredMethod(
+                "createObjectFromFullModuleNames", String.class, Set.class);
+        method.setAccessible(true);
+        final String name = "FooCheck";
+        final String fullName =
+                BASE_PACKAGE + ".packageobjectfactory.bar" + PACKAGE_SEPARATOR + name;
+        final Object instance = method.invoke(factory, name, Collections.singleton(fullName));
+        assertEquals(fullName, instance.getClass().getCanonicalName());
+    }
+
+    @Test
+    public void testCreateObjectFromFullModuleNamesWithException() throws Exception {
+        final Method method = factory.getClass().getDeclaredMethod(
+                "createObjectFromFullModuleNames", String.class, Set.class);
+        method.setAccessible(true);
+        final String name = "FooCheck";
+        final String barFullName =
+                BASE_PACKAGE + ".packageobjectfactory.bar" + PACKAGE_SEPARATOR + name;
+        final String fooFullName =
+                BASE_PACKAGE + ".packageobjectfactory.foo" + PACKAGE_SEPARATOR + name;
+        final Set<String> fullModuleNames = new HashSet<>(Arrays.asList(barFullName, fooFullName));
+        try {
+            method.invoke(factory, name, fullModuleNames);
+            fail("Exception is expected");
+        }
+        catch (InvocationTargetException ex) {
+            final String optionalNames = fullModuleNames.stream()
+                    .reduce((fullNames, fullName) -> fullNames + STRING_SEPARATOR + fullName).get();
+            final LocalizedMessage exceptionMessage = new LocalizedMessage(0,
+                    Definitions.CHECKSTYLE_BUNDLE, AMBIGUOUS_MODULE_NAME_EXCEPTION_MESSAGE,
+                    new String[] {name, optionalNames}, null, getClass(), null);
+            assertEquals(exceptionMessage.getMessage(), ex.getTargetException().getMessage());
+        }
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testGenerateThirdPartyNameToFullModuleName() throws Exception {
+        final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        final String barPackage = BASE_PACKAGE + ".packageobjectfactory.bar";
+        final String fooPackage = BASE_PACKAGE + ".packageobjectfactory.foo";
+        final PackageObjectFactory packageObjectFactory = new PackageObjectFactory(
+                new HashSet<>(Arrays.asList(barPackage, fooPackage)), classLoader);
+        final Method method = packageObjectFactory.getClass().getDeclaredMethod(
+                "generateThirdPartyNameToFullModuleName", ClassLoader.class);
+        method.setAccessible(true);
+        final Map<String, Set<String>> map =
+                (Map<String, Set<String>>) method.invoke(packageObjectFactory, classLoader);
+        map.values().forEach(fullNames -> assertEquals(2, fullNames.size()));
```

Comment:
This test is removed now as it is covered by `testCreateObjectFromFullModuleNamesWithException`